packages/components/view-switcher/src/view-switcher.stories.tsx

```diff
@@ -1,56 +1,40 @@
-import type { ComponentProps } from 'react';
+import type { ReactElement } from 'react';
 import type { Meta, StoryObj } from '@storybook/react';
-import ViewSwitcher from '.';
-import type { TViewSwitcherButtonProps } from './view-switcher-button';
 import { iconArgType, categorize, hideControls } from '@/storybook-helpers';
+import ViewSwitcher, { type TViewSwitcherProps } from '.';
 
 const BASE_CATEGORY = 'View Switcher Button';
 const BUTTONS_COUNT = 4;
 const DEFAULT_BUTTON_ICON = 'WorldIcon';
 
-type ViewSwitcherPropsAndCustomArgs = ComponentProps<
-  typeof ViewSwitcher.Group
-> & {
-  iconViewSwitcherButton1: string;
-  iconViewSwitcherButton2: string;
-  iconViewSwitcherButton3: string;
-  iconViewSwitcherButton4: string;
-  iconOnlyButton1: boolean;
-  iconOnlyButton2: boolean;
-  iconOnlyButton3: boolean;
-  iconOnlyButton4: boolean;
-  isDisabledButton1: boolean;
-  isDisabledButton2: boolean;
-  isDisabledButton3: boolean;
-  isDisabledButton4: boolean;
+type StoryAttributeName =
+  | `iconViewSwitcherButton${number}`
+  | `iconOnlyButton${number}`
+  | `isDisabledButton${number}`;
+type StoryAttributes = {
+  [K in StoryAttributeName]: K extends `iconViewSwitcherButton${number}`
+    ? string
+    : boolean;
 };
+type ViewSwitcherPropsAndCustomArgs = TViewSwitcherProps &
+  StoryAttributes & {
+    children: (index: number) => string;
+  };
 
 const meta = {
   title: 'Components/ViewSwitcher',
   component: ViewSwitcher.Group,
-  tags: ['autodocs'],
```

Comment:
We don't need to specify this in the stories anymore as it has been configured globally in the `.storybook/main.ts` file